src/Middleware/Rewrite/src/UrlMatches/IntegerMatch.cs

```diff
@@ -51,7 +51,7 @@ public override MatchResults Evaluate(string input, RewriteContext context)
                 case IntegerOperationType.NotEqual:
                     return compValue != _value ? MatchResults.EmptySuccess : MatchResults.EmptyFailure;
                 default:
-                    throw new ArgumentOutOfRangeException("operation"); // Will never be thrown
+                    throw new ArgumentOutOfRangeException(nameof(input)); // Will never be thrown
```

Comment:
It's not switching on input. Maybe InvalidOperationException? Or move the validation to the constructor?